### PR TITLE
Add access control logic for request creators

### PR DESCRIPTION
**Pull request by loldot**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
- Add new auth requirement for request creator
- Implement security matrix for request creator

[Security Matrix](https://teams.microsoft.com/l/file/447EDF2B-9A6F-4163-A77E-E896A98E0617?tenantId=3aa4a235-b6e2-48d5-9195-7fcf05b459b0&fileType=xlsx&objectUrl=https%3A%2F%2Fstatoilsrm.sharepoint.com%2Fsites%2FProView190%2FShared%20Documents%2FFeatures%20-%20Resources%2FResource%20allocation%2FFusion-Resource-Allocation-API.xlsx&baseUrl=https%3A%2F%2Fstatoilsrm.sharepoint.com%2Fsites%2FProView190&serviceName=teams&threadId=19:1382bf30a0614a6b86a0a15c55a9833b@thread.skype&groupId=33a6ebec-c93b-47ae-94b9-fbc6fa6c4a4b)

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

**Checklist:**
- [x] Considered automated tests
- [x] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs [hunk 1]
```diff
@@ -100,7 +99,7 @@ public class InternalRequestsController : ResourceControllerBase
                 r.AlwaysAccessWhen().FullControl().FullControlInternal();
                 r.AnyOf(or =>
                 {
-
+                    or.BeResourceOwner(departmentPath, includeParents: false, includeDescendants: true);
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs:102]: Should maybe also validate that the relevant resource on the instance exists in the department?

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs [hunk 2]
```diff
@@ -187,6 +186,10 @@ public class InternalRequestsController : ResourceControllerBase
 
                     if (item.AssignedDepartment is null && item.OrgPosition is not null)
                         or.BeResourceOwner(new DepartmentPath(item.OrgPosition.BasePosition.Department).GoToLevel(3), includeDescendants: true);
+
+                    if (!request.AssignedDepartment.HasValue && !request.ProposedPersonAzureUniqueId.HasValue)
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs:190]: The request creator is not really dependant on any props.. it should be added as a default check..

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs [hunk 3]
```diff
@@ -487,6 +492,8 @@ public async Task<ActionResult> DeleteAllocationRequest(Guid requestId)
                 {
                     if (result.Type == InternalRequestType.Allocation)
                     {
+                        or.BeRequestCreator(requestId);
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs:495]: Should add BeRequestCreator to starting resource owner requests aswell? (can't comment on lines not changed, so adding here :) )

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs [hunk 4]
```diff
@@ -573,6 +580,7 @@ public async Task<ActionResult<ApiResourceAllocationRequest>> ApproveProjectAllo
                 {
                     if (result.OrgPositionId.HasValue)
                         or.OrgChartPositionWriteAccess(result.Project.OrgProjectId, result.OrgPositionId.Value);
+                    or.BeRequestCreator(requestId);
```
Comment (HansDahle) [inline src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs:583]: Hmm, it kind of make sense, but it kind of doesnt.. I think we might have to offload this to the logic? 
Just seem like we need to take into account what type the workflow is.

Ex. now i can do the "propose" from owner side by hitting this endpoint, and if i created the request as a task owner, i can also propose.. I think these approve steps must be delt with separately.

**Review by HansDahle — COMMENTED** [review]
Nice, but i think there is something around the approval/logic steps we need to handle differently.

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs (hunk 2 above)
Comment (loldot) [reply src/backend/api/Fusion.Resources.Api/Controllers/Requests/InternalRequestsController.cs:190]: It was meant to handle these lines from the excel sheet. Should the excel sheet be updated or should they be implemented differently?
- Reassign to department
- Propose

**Review by HansDahle — APPROVED** [review]

**Pull request merged**